packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -229,6 +229,51 @@ export enum SourceControlInputBoxValidationType {
     Information = 2
 }
 
+/**
+ * Details if an `ExternalUriOpener` can open a uri.
+ *
+ * The priority is also used to rank multiple openers against each other and determine
+ * if an opener should be selected automatically or if the user should be prompted to
+ * select an opener.
+ *
+ * The editor will try to use the best available opener, as sorted by `ExternalUriOpenerPriority`.
+ * If there are multiple potential "best" openers for a URI, then the user will be prompted
+ * to select an opener.
+ */
+export enum ExternalUriOpenerPriority {
+    /**
+     * The opener is disabled and will never be shown to users.
+     *
+     * Note that the opener can still be used if the user specifically
+     * configures it in their settings.
+     */
+    None = 0,
+
+    /**
+     * The opener can open the uri but will not cause a prompt on its own
+     * since the editor always contributes a built-in `Default` opener.
+     */
+    Option = 1,
+
+    /**
+     * The opener can open the uri.
+     *
+     * The editor's built-in opener has `Default` priority. This means that any additional `Default`
+     * openers will cause the user to be prompted to select from a list of all potential openers.
+     */
+    Default = 2,
+
+    /**
+     * The opener can open the uri and should be automatically selected over any
+     * default openers, include the built-in one from the editor.
+     *
+     * A preferred opener will be automatically selected if no other preferred openers
+     * are available. If multiple preferred openers are available, then the user
+     * is shown a prompt with all potential openers (not just preferred openers).
+     */
+    Preferred = 3,
+}
```

Comment:
minor: we can probably remove the documentation given it is just duplication from `packages/plugin/src/theia.proposed.externalUriOpener.d.ts` anyways and simplifies the changes.